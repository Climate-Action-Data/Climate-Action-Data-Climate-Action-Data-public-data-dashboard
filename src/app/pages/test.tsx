'use client'
import { type NextPage } from "next";

const Test: NextPage = () => {
    return (
        <div>
            <h1>Hello</h1>
            <p>data</p>
        </div>
    )
}

export default Test