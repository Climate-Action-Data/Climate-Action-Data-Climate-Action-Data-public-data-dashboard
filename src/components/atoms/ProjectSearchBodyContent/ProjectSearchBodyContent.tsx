import { changeHoverColor } from '@/utils/Stickify'
import { Table, Tbody, Td, Tr } from '@chakra-ui/react'
import { ProjectSearchResponse, ProjectSearchResult } from '@/@types/ProjectSearchResult'
import { EffectResponse } from '@/@types/EffectResponse'
import { useRouter } from 'next/navigation'
import { ProjectSearchBodySkeleton } from '../ProjectSearchBodySkeleton/ProjectSearchBodySkeleton'
import { formatDate } from '@/utils/DateFormat'
import { DateFormats } from '@/@types/DateFormats'
import { TableData } from '../TableData/TableData'
const renderCreditingPeriod = (startDate: string | undefined, endDate: string | undefined) => {
  if (startDate && endDate) {
    return `${formatDate(startDate, DateFormats.YYYY_MM_DD)} - ${formatDate(endDate, DateFormats.YYYY_MM_DD)}`
  }
  if (startDate) {
    return formatDate(startDate, DateFormats.YYYY_MM_DD)
  }
  if (endDate) {
    return formatDate(endDate, DateFormats.YYYY_MM_DD)
  }
}

interface ProjectSearchBodyContentProps {
  projectResults?: EffectResponse<ProjectSearchResponse>
}

export const ProjectSearchBodyContent = (props: ProjectSearchBodyContentProps) => {
  const { projectResults } = props
  const router = useRouter()

  const generateTableRow = (projectList: ProjectSearchResult[]) => {
    return projectList.map((project, idx) => (
      <Tr
        onClick={() => router.push(`/project/${project.warehouseProjectId}`)}
        data-testid="table-row"
        onMouseEnter={() => changeHoverColor(`project-row-${idx}`, `hoverGreen`)}
        className={`project-row-${idx}`}
        key={`project-body-row-${project.id}`}
        height="92px"
      >
        <Td title={project.standard}>
          <TableData data={project.standard} />
        </Td>
        <Td title={project.methodology}>
          <TableData data={project.methodology} />
        </Td>
        <Td title={project.sector}>
          <TableData data={project.sector} />
        </Td>
        <Td title={project.country}>
          <TableData data={project.country} />
        </Td>
        <Td title={project.status}>
          <TableData data={project.status} />
        </Td>
        <Td minW="250px !important" title={`${project.creditingPeriodStart}-${project.creditingPeriodEnd}`}>
          <TableData data={renderCreditingPeriod(project.creditingPeriodStart, project.creditingPeriodEnd)} />
        </Td>
        <Td isNumeric title={project.annualEst?.toString()}>
          <TableData data={project.annualEst} />
        </Td>
        <Td isNumeric title={project.annualIssued?.toString()}>
          <TableData data={project.annualIssued} />
        </Td>
        <Td isNumeric title={project.annualRetired?.toString()}>
          <TableData data={project.annualRetired} />
        </Td>
        <Td isNumeric title={project.annualAvailable?.toString()}>
          <TableData data={project.annualAvailable} />
        </Td>
      </Tr>
    ))
  }

  return (
    <Table variant="simple" className="searchTable">
      <Tbody data-testid="table-scroll">{projectResults?.data?.projects ? generateTableRow(projectResults.data.projects) : <ProjectSearchBodySkeleton />}</Tbody>
    </Table>
  )
}
