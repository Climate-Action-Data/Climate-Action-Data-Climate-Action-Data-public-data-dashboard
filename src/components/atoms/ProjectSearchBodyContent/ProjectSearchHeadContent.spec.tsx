import { fireEvent, render, screen } from '@testing-library/react'

import { ProjectSearchBodyContent } from './ProjectSearchBodyContent'
import { PROJECT_SEARCH_RESULT } from '@/test/TestOvermindMockData'
import { TestRouter } from '@/test/TestRouter'

it(`renders correctly`, () => {
  const { container } = render(
    <TestRouter router={{}}>
      <ProjectSearchBodyContent projectResults={PROJECT_SEARCH_RESULT.projectResults} />
    </TestRouter>,
  )
  expect(container).toMatchSnapshot()
})

it(`renders correctly with hover`, () => {
  const { container } = render(
    <TestRouter router={{}}>
      <ProjectSearchBodyContent projectResults={PROJECT_SEARCH_RESULT.projectResults} />
    </TestRouter>,
  )
  const rows = screen.getAllByTestId(`table-row`)
  expect(rows.length).toBeGreaterThan(0)
  fireEvent.mouseOver(rows[0])
  expect(container).toMatchSnapshot()
})
