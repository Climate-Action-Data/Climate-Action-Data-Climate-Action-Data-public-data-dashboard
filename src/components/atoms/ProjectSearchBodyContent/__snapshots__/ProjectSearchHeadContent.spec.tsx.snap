// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <table
    class="chakra-table searchTable css-0"
  >
    <tbody
      class="css-0"
      data-testid="table-scroll"
    >
      <tr
        class="project-row-0 css-mii0oe"
        data-testid="table-row"
      >
        <td
          class="css-0"
          title="EcoRegistry"
        >
          EcoRegistry
        </td>
        <td
          class="css-0"
          title="ACM0002: Grid-connected electricity generation from renewable sources / ACM0006: Electricity and heat generation from biomass "
        >
          ACM0002: Grid-connected electricity generation from renewable sources / ACM0006: Electricity and heat generation from biomass 
        </td>
        <td
          class="css-0"
          title="Electricity; gas, steam and air conditioning supply"
        >
          Electricity; gas, steam and air conditioning supply
        </td>
        <td
          class="css-0"
          title="Colombia"
        >
          Colombia
        </td>
        <td
          class="css-0"
          title="Authorized"
        >
          Authorized
        </td>
        <td
          class="css-rl3ocm"
          title="2020-08-22T00:00:00.000Z-2030-08-21T00:00:00.000Z"
        >
          2020/08/22 - 2030/08/21
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="311410"
        >
          311,410
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="37202"
        >
          37,202
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="30167"
        >
          30,167
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="7035"
        >
          7,035
        </td>
      </tr>
      <tr
        class="project-row-1 css-mii0oe"
        data-testid="table-row"
      >
        <td
          class="css-0"
          title="EcoRegistry"
        >
          EcoRegistry
        </td>
        <td
          class="css-0"
          title="AR-ACM0003: Afforestation and reforestation of lands except wetlands"
        >
          AR-ACM0003: Afforestation and reforestation of lands except wetlands
        </td>
        <td
          class="css-0"
          title="Agriculture; forestry and fishing"
        >
          Agriculture; forestry and fishing
        </td>
        <td
          class="css-0"
          title="Colombia"
        >
          Colombia
        </td>
        <td
          class="css-0"
          title="Authorized"
        >
          Authorized
        </td>
        <td
          class="css-rl3ocm"
          title="2010-05-02T00:00:00.000Z-undefined"
        >
          2010/05/02
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="128161"
        >
          128,161
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="225734"
        >
          225,734
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="147403"
        >
          147,403
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="78331"
        >
          78,331
        </td>
      </tr>
      <tr
        class="project-row-2 css-mii0oe"
        data-testid="table-row"
      >
        <td
          class="css-0"
          title="EcoRegistry"
        >
          EcoRegistry
        </td>
        <td
          class="css-0"
          title="M/LU-REDD+: Methodology for the Implementation of REDD+ Projects Consistent with National Reference Levels"
        >
          M/LU-REDD+: Methodology for the Implementation of REDD+ Projects Consistent with National Reference Levels
        </td>
        <td
          class="css-0"
          title="Agriculture; forestry and fishing"
        >
          Agriculture; forestry and fishing
        </td>
        <td
          class="css-0"
          title="Colombia"
        >
          Colombia
        </td>
        <td
          class="css-0"
          title="Approved"
        >
          Approved
        </td>
        <td
          class="css-rl3ocm"
          title="undefined-2044-05-05T00:00:00.000Z"
        >
          2044/05/05
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="35613"
        >
          35,613
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="0"
        >
          0
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="0"
        >
          0
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="0"
        >
          0
        </td>
      </tr>
      <tr
        class="project-row-3 css-mii0oe"
        data-testid="table-row"
      >
        <td
          class="css-0"
          title="EcoRegistry"
        >
          EcoRegistry
        </td>
        <td
          class="css-0"
          title="M/LU-REDD+: Methodology for the Implementation of REDD+ Projects Consistent with National Reference Levels"
        >
          M/LU-REDD+: Methodology for the Implementation of REDD+ Projects Consistent with National Reference Levels
        </td>
        <td
          class="css-0"
          title="Agriculture; forestry and fishing"
        >
          Agriculture; forestry and fishing
        </td>
        <td
          class="css-0"
          title="Colombia"
        >
          Colombia
        </td>
        <td
          class="css-0"
          title="Authorized"
        >
          Authorized
        </td>
        <td
          class="css-rl3ocm"
          title="2017-11-01T00:00:00.000Z-2027-10-31T00:00:00.000Z"
        >
          2017/11/01 - 2027/10/31
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="357571"
        >
          357,571
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="172795"
        >
          172,795
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="0"
        >
          0
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="172795"
        >
          172,795
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`renders correctly 2`] = `
<div>
  <table
    class="chakra-table searchTable css-0"
  >
    <tbody
      class="css-0"
      data-testid="table-scroll"
    >
      <tr
        class="project-row-0 css-mii0oe"
        data-testid="table-row"
      >
        <td
          class="css-0"
          title="EcoRegistry"
        >
          EcoRegistry
        </td>
        <td
          class="css-0"
          title="ACM0002: Grid-connected electricity generation from renewable sources / ACM0006: Electricity and heat generation from biomass "
        >
          ACM0002: Grid-connected electricity generation from renewable sources / ACM0006: Electricity and heat generation from biomass 
        </td>
        <td
          class="css-0"
          title="Electricity; gas, steam and air conditioning supply"
        >
          Electricity; gas, steam and air conditioning supply
        </td>
        <td
          class="css-0"
          title="Colombia"
        >
          Colombia
        </td>
        <td
          class="css-0"
          title="Authorized"
        >
          Authorized
        </td>
        <td
          class="css-rl3ocm"
          title="2020-08-22T00:00:00.000Z-2030-08-21T00:00:00.000Z"
        >
          2020/08/22 - 2030/08/21
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="311410"
        >
          311,410
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="37202"
        >
          37,202
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="30167"
        >
          30,167
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="7035"
        >
          7,035
        </td>
      </tr>
      <tr
        class="project-row-1 css-mii0oe"
        data-testid="table-row"
      >
        <td
          class="css-0"
          title="EcoRegistry"
        >
          EcoRegistry
        </td>
        <td
          class="css-0"
          title="AR-ACM0003: Afforestation and reforestation of lands except wetlands"
        >
          AR-ACM0003: Afforestation and reforestation of lands except wetlands
        </td>
        <td
          class="css-0"
          title="Agriculture; forestry and fishing"
        >
          Agriculture; forestry and fishing
        </td>
        <td
          class="css-0"
          title="Colombia"
        >
          Colombia
        </td>
        <td
          class="css-0"
          title="Authorized"
        >
          Authorized
        </td>
        <td
          class="css-rl3ocm"
          title="2010-05-02T00:00:00.000Z-undefined"
        >
          2010/05/02
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="128161"
        >
          128,161
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="225734"
        >
          225,734
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="147403"
        >
          147,403
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="78331"
        >
          78,331
        </td>
      </tr>
      <tr
        class="project-row-2 css-mii0oe"
        data-testid="table-row"
      >
        <td
          class="css-0"
          title="EcoRegistry"
        >
          EcoRegistry
        </td>
        <td
          class="css-0"
          title="M/LU-REDD+: Methodology for the Implementation of REDD+ Projects Consistent with National Reference Levels"
        >
          M/LU-REDD+: Methodology for the Implementation of REDD+ Projects Consistent with National Reference Levels
        </td>
        <td
          class="css-0"
          title="Agriculture; forestry and fishing"
        >
          Agriculture; forestry and fishing
        </td>
        <td
          class="css-0"
          title="Colombia"
        >
          Colombia
        </td>
        <td
          class="css-0"
          title="Approved"
        >
          Approved
        </td>
        <td
          class="css-rl3ocm"
          title="undefined-2044-05-05T00:00:00.000Z"
        >
          2044/05/05
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="35613"
        >
          35,613
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="0"
        >
          0
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="0"
        >
          0
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="0"
        >
          0
        </td>
      </tr>
      <tr
        class="project-row-3 css-mii0oe"
        data-testid="table-row"
      >
        <td
          class="css-0"
          title="EcoRegistry"
        >
          EcoRegistry
        </td>
        <td
          class="css-0"
          title="M/LU-REDD+: Methodology for the Implementation of REDD+ Projects Consistent with National Reference Levels"
        >
          M/LU-REDD+: Methodology for the Implementation of REDD+ Projects Consistent with National Reference Levels
        </td>
        <td
          class="css-0"
          title="Agriculture; forestry and fishing"
        >
          Agriculture; forestry and fishing
        </td>
        <td
          class="css-0"
          title="Colombia"
        >
          Colombia
        </td>
        <td
          class="css-0"
          title="Authorized"
        >
          Authorized
        </td>
        <td
          class="css-rl3ocm"
          title="2017-11-01T00:00:00.000Z-2027-10-31T00:00:00.000Z"
        >
          2017/11/01 - 2027/10/31
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="357571"
        >
          357,571
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="172795"
        >
          172,795
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="0"
        >
          0
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="172795"
        >
          172,795
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`renders correctly with hover 1`] = `
<div>
  <table
    class="chakra-table searchTable css-0"
  >
    <tbody
      class="css-0"
      data-testid="table-scroll"
    >
      <tr
        class="project-row-0 css-mii0oe hoverGreen"
        data-testid="table-row"
      >
        <td
          class="css-0"
          title="EcoRegistry"
        >
          EcoRegistry
        </td>
        <td
          class="css-0"
          title="ACM0002: Grid-connected electricity generation from renewable sources / ACM0006: Electricity and heat generation from biomass "
        >
          ACM0002: Grid-connected electricity generation from renewable sources / ACM0006: Electricity and heat generation from biomass 
        </td>
        <td
          class="css-0"
          title="Electricity; gas, steam and air conditioning supply"
        >
          Electricity; gas, steam and air conditioning supply
        </td>
        <td
          class="css-0"
          title="Colombia"
        >
          Colombia
        </td>
        <td
          class="css-0"
          title="Authorized"
        >
          Authorized
        </td>
        <td
          class="css-rl3ocm"
          title="2020-08-22T00:00:00.000Z-2030-08-21T00:00:00.000Z"
        >
          2020/08/22 - 2030/08/21
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="311410"
        >
          311,410
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="37202"
        >
          37,202
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="30167"
        >
          30,167
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="7035"
        >
          7,035
        </td>
      </tr>
      <tr
        class="project-row-1 css-mii0oe"
        data-testid="table-row"
      >
        <td
          class="css-0"
          title="EcoRegistry"
        >
          EcoRegistry
        </td>
        <td
          class="css-0"
          title="AR-ACM0003: Afforestation and reforestation of lands except wetlands"
        >
          AR-ACM0003: Afforestation and reforestation of lands except wetlands
        </td>
        <td
          class="css-0"
          title="Agriculture; forestry and fishing"
        >
          Agriculture; forestry and fishing
        </td>
        <td
          class="css-0"
          title="Colombia"
        >
          Colombia
        </td>
        <td
          class="css-0"
          title="Authorized"
        >
          Authorized
        </td>
        <td
          class="css-rl3ocm"
          title="2010-05-02T00:00:00.000Z-undefined"
        >
          2010/05/02
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="128161"
        >
          128,161
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="225734"
        >
          225,734
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="147403"
        >
          147,403
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="78331"
        >
          78,331
        </td>
      </tr>
      <tr
        class="project-row-2 css-mii0oe"
        data-testid="table-row"
      >
        <td
          class="css-0"
          title="EcoRegistry"
        >
          EcoRegistry
        </td>
        <td
          class="css-0"
          title="M/LU-REDD+: Methodology for the Implementation of REDD+ Projects Consistent with National Reference Levels"
        >
          M/LU-REDD+: Methodology for the Implementation of REDD+ Projects Consistent with National Reference Levels
        </td>
        <td
          class="css-0"
          title="Agriculture; forestry and fishing"
        >
          Agriculture; forestry and fishing
        </td>
        <td
          class="css-0"
          title="Colombia"
        >
          Colombia
        </td>
        <td
          class="css-0"
          title="Approved"
        >
          Approved
        </td>
        <td
          class="css-rl3ocm"
          title="undefined-2044-05-05T00:00:00.000Z"
        >
          2044/05/05
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="35613"
        >
          35,613
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="0"
        >
          0
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="0"
        >
          0
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="0"
        >
          0
        </td>
      </tr>
      <tr
        class="project-row-3 css-mii0oe"
        data-testid="table-row"
      >
        <td
          class="css-0"
          title="EcoRegistry"
        >
          EcoRegistry
        </td>
        <td
          class="css-0"
          title="M/LU-REDD+: Methodology for the Implementation of REDD+ Projects Consistent with National Reference Levels"
        >
          M/LU-REDD+: Methodology for the Implementation of REDD+ Projects Consistent with National Reference Levels
        </td>
        <td
          class="css-0"
          title="Agriculture; forestry and fishing"
        >
          Agriculture; forestry and fishing
        </td>
        <td
          class="css-0"
          title="Colombia"
        >
          Colombia
        </td>
        <td
          class="css-0"
          title="Authorized"
        >
          Authorized
        </td>
        <td
          class="css-rl3ocm"
          title="2017-11-01T00:00:00.000Z-2027-10-31T00:00:00.000Z"
        >
          2017/11/01 - 2027/10/31
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="357571"
        >
          357,571
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="172795"
        >
          172,795
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="0"
        >
          0
        </td>
        <td
          class="css-0"
          data-is-numeric="true"
          title="172795"
        >
          172,795
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
